subsys/net/openthread/rpc/client/ot_rpc_coap_client.c

```diff
@@ -0,0 +1,316 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <ot_rpc_ids.h>
+#include <ot_rpc_common.h>
+#include <nrf_rpc/nrf_rpc_serialize.h>
+
+#include <nrf_rpc_cbor.h>
+
+#include <openthread/coap.h>
+
+static otCoapResource * resources;
+static otCoapRequestHandler default_handler;
+static void *default_handler_ctx;
+
+otMessage *otCoapNewMessage(otInstance *aInstance, const otMessageSettings *aSettings)
+{
+	struct nrf_rpc_cbor_ctx ctx;
+	uintptr_t message_rep;
+
+	NRF_RPC_CBOR_ALLOC(&ot_group, ctx, OT_RPC_MESSAGE_SETTINGS_LENGTH);
+	ot_rpc_encode_message_settings(&ctx, aSettings);
+
+	nrf_rpc_cbor_cmd_rsp_no_err(&ot_group, OT_RPC_CMD_COAP_NEW_MESSAGE, &ctx);
+	nrf_rpc_rsp_decode_uint(&ot_group, &ctx, &message_rep, sizeof(message_rep));
+	nrf_rpc_cbor_decoding_done(&ot_group, &ctx);
+
+	return (otMessage *)message_rep;
+}
+
+void otCoapMessageInit(otMessage *aMessage, otCoapType aType, otCoapCode aCode)
+{
+	struct nrf_rpc_cbor_ctx ctx;
+	size_t cbor_buffer_size = 0;
+
+	cbor_buffer_size += 1 + sizeof(uintptr_t); /* aMessage */
+	cbor_buffer_size += 1;			   /* aType */
+	cbor_buffer_size += 1 + sizeof(aCode);	   /* aCode */
+
+	NRF_RPC_CBOR_ALLOC(&ot_group, ctx, cbor_buffer_size);
+	nrf_rpc_encode_uint(&ctx, (uintptr_t)aMessage);
+	nrf_rpc_encode_uint(&ctx, aType);
+	nrf_rpc_encode_uint(&ctx, aCode);
+
+	nrf_rpc_cbor_cmd_no_err(&ot_group, OT_RPC_CMD_COAP_MESSAGE_INIT, &ctx,
+				nrf_rpc_rsp_decode_void, NULL);
+}
+
+otError otCoapMessageInitResponse(otMessage *aResponse, const otMessage *aRequest, otCoapType aType,
+				  otCoapCode aCode)
+{
+	struct nrf_rpc_cbor_ctx ctx;
+	size_t cbor_buffer_size = 0;
+	otError error;
+
+	cbor_buffer_size += 1 + sizeof(uintptr_t); /* aResponse */
+	cbor_buffer_size += 1 + sizeof(uintptr_t); /* aRequest */
+	cbor_buffer_size += 1;			   /* aType */
+	cbor_buffer_size += 1 + sizeof(aCode);	   /* aCode */
+
+	NRF_RPC_CBOR_ALLOC(&ot_group, ctx, cbor_buffer_size);
+	nrf_rpc_encode_uint(&ctx, (uintptr_t)aResponse);
+	nrf_rpc_encode_uint(&ctx, (uintptr_t)aRequest);
+	nrf_rpc_encode_uint(&ctx, aType);
+	nrf_rpc_encode_uint(&ctx, aCode);
+
+	nrf_rpc_cbor_cmd_no_err(&ot_group, OT_RPC_CMD_COAP_MESSAGE_INIT_RESPONSE, &ctx,
+				ot_rpc_decode_error, &error);
+
+	return error;
+}
+
+otError otCoapMessageAppendUriPathOptions(otMessage *aMessage, const char *aUriPath)
+{
+	struct nrf_rpc_cbor_ctx ctx;
+	size_t cbor_buffer_size = 0;
+	otError error;
+
+	cbor_buffer_size += 1 + sizeof(uintptr_t); /* aMessage */
```

Comment:
Please use `ot_msg_key` instedo of `uintptr_t`.

Reply:
I assumed that this is implementation detail of the server and the client should assume this is just `otMessage*` being sent but I can change it.